src/map/skill.cpp

```diff
@@ -19191,11 +19191,7 @@ bool skill_check_condition_castbegin( map_session_data& sd, uint16 skill_id, uin
 			}
 			break;
 		case ST_RECOVER_WEIGHT_RATE:
-#ifdef RENEWAL
-			if(pc_is70overweight(&sd)) {
-#else
-			if(pc_is50overweight(&sd)) {
-#endif
+			if( pc_getpercentweight(sd) >= battle_config.natural_heal_weight_rate ) {
```

Comment:
Should use SC_WEIGHT50 check?

Reply:
I've changed it to check SC_WEIGHT50 and SC_WEIGHT90, but maybe we should check sd.regen.state.overweight instead?

Reply:
Would make sense!